core/src/utils/content/index.ts

```diff
@@ -0,0 +1,130 @@
+import { componentOnReady } from '../helpers';
+import { printRequiredElementError } from '../logging';
+
+const ION_CONTENT_ELEMENT_SELECTOR = 'ion-content';
+const ION_CONTENT_CLASS_SELECTOR = '.ion-content-scroll-host';
+/**
+ * Selector used for implementations reliant on `<ion-content>` for scroll event changes.
+ *
+ * Developers should use the `.ion-content-scroll-host` selector to target the element emitting
+ * scroll events. With virtual scroll implementations this will be the host element for
+ * the scroll viewport.
+ */
+const ION_CONTENT_SELECTOR = `${ION_CONTENT_ELEMENT_SELECTOR}, ${ION_CONTENT_CLASS_SELECTOR}`;
+
+const hasOverflowScroll = (node: Element | null) => {
+  if (node === null) {
+    return false;
+  }
+  const isElement = node instanceof HTMLElement;
+  const overflowY = isElement && getComputedStyle(node).overflowY;
+  // Element is scrollable if there is overflow or if the overlay is explicit
+  const isScrollable = overflowY !== 'visible' && overflowY !== 'hidden';
+  // The element is the node if it's scrollable or the scroll height is larger than the client height
+  if (isScrollable && node.scrollHeight >= node.clientHeight) {
+    return true;
+  }
+  return false;
+}
+
+const getScrollContainer = (node: Element): HTMLElement | null => {
+  if (hasOverflowScroll(node)) {
+    return node as HTMLElement;
+  }
+  for (const item of Array.from(node.children)) {
+    if (item && hasOverflowScroll(item)) {
+      return item as HTMLElement;
+    }
+  }
+  return null;
+}
+
+/**
+ * Waits for the element host fully initialize before
+ * returning the inner scroll element.
+ *
+ * For `ion-content` the scroll target will be the result
+ * of the `getScrollElement` function.
+ *
+ * For custom implementations it will be the element host
+ * or a selector within the host, if supplied through `scrollTarget`.
+ */
+export const getScrollElement = async (el: Element) => {
+  await new Promise(resolve => componentOnReady(el, resolve));
+
+  if (el?.tagName.toLowerCase() === ION_CONTENT_ELEMENT_SELECTOR) {
```

Comment:
This check is very explicit to calling custom methods exposed on `ion-content`. I do not think we would want to open that up to other selectors, otherwise it will throw exceptions.